repo/mk/js.deps.npm.mk

```diff
@@ -79,3 +76,22 @@ deps-npm-prod:
 		$(NPM) update --no-save --production; \
 	}
 	$(NPM) list --depth=0 || $(MAKE) deps-npm-unmet-peer
+
+
+# the audit can be ignored by adding to the Makefile:
+# SF_DEPS_TARGETS := $(filter-out deps-npm-audit,$(SF_DEPS_TARGETS))
+
+.PHONY: deps-npm-audit
+deps-npm-audit:
+	$(ECHO_DO) "Auditing package.json..."
+	$(GIT) ls-files --error-unmatch "package-lock.json" || {
```

Comment:
What's the idea behind this line?

Reply:
it's a way to check if `package-lock.json` is checked in
we generally do not check it in (due to the .npmrc and .gitignore config), but there's nothing stopping someone from doing so

thing is that `npm audit` can only be run if there is a package-lock.json. the longer story is that it is actually run automatically by `npm install` (but it doesn't exit with a non-zero code)

so what we need to do is first create a `package-lock.json`, if one isn't checked-in
then we run `npm audit` and maybe `npm audit --fix`
and then we maybe clean up the intermediary `package-lock.json` artifact

Reply:
Thanks for explanation!

Reply:
added a bit of info on that as visual cues